roachpb/batch.go

```diff
@@ -176,11 +176,264 @@ func (ba *BatchRequest) Methods() []Method {
 }
 
 // CreateReply creates replies for each of the contained requests, wrapped in a
-// BatchResponse.
+// BatchResponse. The response objects are batch allocated to minimize
+// allocation overhead which adds a bit of complexity to this function.
 func (ba *BatchRequest) CreateReply() *BatchResponse {
 	br := &BatchResponse{}
+	br.Responses = make([]ResponseUnion, len(ba.Requests))
+
+	var counts struct {
```

Comment:
Yeah, can't imagine anything that that wouldn't already catch. SGTM

On Fri, Apr 22, 2016 at 9:29 AM Peter Mattis <notifications@github.com>
wrote:

> In roachpb/batch.go
> <https://github.com/cockroachdb/cockroach/pull/6222#discussion_r60738294>:
>
> >  func (ba *BatchRequest) CreateReply() *BatchResponse {
> >  	br := &BatchResponse{}
> > +	br.Responses = make([]ResponseUnion, len(ba.Requests))
> > +
> > +	var counts struct {
>
> I considered this while writing this PR, but notice that if you add a new
> request/response to Batch{Request,Response} and don't populate it here
> you'll get a panic when trying to implement it. I discovered this when I
> accidentally left out Increment{Request,Response}. Another easy error
> here which is caught is incrementing the wrong count or using the wrong
> count when constructing the buffer. But as long as all of the requests are
> tested by other code, we'll hit panics when trying to populate the
> responses.
>
> Push-back if you see a type of error here that wouldn't be caught be
> existing tests.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6222/files/761673a2b06ec8203033d3b1213cdc90a5d8459b#r60738294>
>
-- 

-- Tobias